src/main/java/com/minecolonies/event/EventHandler.java

```diff
@@ -63,22 +111,23 @@ public void onBlockBreak(@NotNull BlockEvent.BreakEvent event)
      * Event gets cancelled when player has no permission
      * Event gets cancelled when the player has no permission to place a hut, and tried it
      *
-     * @param event {@link PlayerInteractEvent}
+     * @param event {@link PlayerInteractEvent.RightClickBlock}
      */
     @SubscribeEvent
-    public void onPlayerInteract(@NotNull PlayerInteractEvent event)
+    public void onPlayerInteract(@NotNull PlayerInteractEvent.RightClickBlock event)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "onPlayerInteract" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)